MapboxNavigation/RouteTableViewController.swift

```diff
@@ -68,20 +68,43 @@ class RouteTableViewController: UIViewController {
 extension RouteTableViewController: UITableViewDelegate, UITableViewDataSource {
 
     func numberOfSections(in tableView: UITableView) -> Int {
-        return 1
+        return routeController.routeProgress.route.legs.count
+    }
+    
+    func tableView(_ tableView: UITableView, titleForHeaderInSection section: Int) -> String? {
+        // Don't display section header if there is only one step
+        guard routeController.routeProgress.route.legs.count > 1 else {
+            return nil
+        }
+        let leg = routeController.routeProgress.route.legs[section]
+        
+        let sourceName = leg.source.name
+        let destinationName = leg.destination.name
+        
+        if let sourceName = sourceName?.nonEmptyString, let destinationName = destinationName?.nonEmptyString {
```

Comment:
If the source is the user’s unnamed current location and the destination is a named location, we should say “Destination via Street Name and Street Name”.

If the source is a named location and the destination is unnamed for some reason, we should say “Street Name and Street Name”, same as below.